framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -1572,17 +1591,36 @@ AutomaticMortarGeneration::computeNodalGeometry()
     _secondary_node_to_hh_nodal_tangents[_mesh.node_ptr(node_id)][1] = nodal_tangent_two;
   }
 }
+
 void
 AutomaticMortarGeneration::householderOrthogolization(const Point & nodal_normal,
                                                       Point & nodal_tangent_one,
                                                       Point & nodal_tangent_two) const
 {
-  Point unit_normal = nodal_normal.unit();
+  mooseAssert(MooseUtils::absoluteFuzzyEqual(nodal_normal.norm(), 1),
+              "The input nodal normal should have unity norm");
+
+  const Real nx = nodal_normal(0);
+  const Real ny = nodal_normal(1);
+  const Real nz = nodal_normal(2);
+
   // See Lopes DS, Silva MT, Ambrosio JA. Tangent vectors to a 3-D surface normal: A geometric tool
   // to find orthogonal vectors based on the Householder transformation. Computer-Aided Design. 2013
   // Mar 1;45(3):683-94.
-  const Point h_vector(
-      std::max(unit_normal(0) - 1.0, unit_normal(0) + 1.0), unit_normal(1), unit_normal(2));
+  const Point h_vector(std::max(nx - 1.0, nx + 1.0), ny, nz);
+
+  if (std::abs(h_vector(0)) < TOLERANCE)
+  {
+    nodal_tangent_one(0) = 0;
+    nodal_tangent_one(1) = 1;
+    nodal_tangent_one(2) = 0;
+
+    nodal_tangent_two(0) = 0;
+    nodal_tangent_two(1) = 0;
+    nodal_tangent_two(2) = -1;
+
+    return;
+  }
```

Comment:
I assume there was a methodology to this choice? If it's not too complicated, you might consider adding a comment regarding it

Reply:
Yeah, the equations yield these vectors as we approach the singularity. Will add a comment to that respect.

Reply:
This is an example of how the fields look like on a spherical surface for a node that doesn't go in the avoid-singularity if statement:

<img width="1857" alt="image" src="https://user-images.githubusercontent.com/14300313/178068523-2ef091d5-4e4f-46df-b546-351ecc7cd0ea.png">